### DISCUSSION TITLE
Download the videos of the top 10 posts from a subreddit

### DISCUSSION THREAD

**Discussion by music-lover99** [elmoiv/redvid#28] [Q&A]
I was wondering how I could download the videos from the top 10 posts of a subreddit.

The program would go through the top posts of a subreddit, if there isn't a video it would skip the post, and if there is a video it would download it (I'd end up with 10 .mp4 files)

--------------------------------

Also, is there a way to download all the videos from a subreddit in a loop? Something following the lines of this:

```
#doesn't follow any syntax, it's just to illustrate what I'm saying

subreddit_posts = subreddit("python")
for post in subreddit_posts:
    if post.contains("video"):
        download(post.video)
    else:
         pass
```

**Comment by elmoiv** [comment]
You need first to consider that not all top posts are videos. They may be pictures or just text. 
You need to do the following:
- Use a Pythonic API for Reddit to fetch these posts like [PRAW](https://github.com/praw-dev/praw) or do it yourself.
- Filter video posts from results.
- Pass each video to a `redvid` instance and start downloading.
